src/main/java/com/fasterxml/jackson/databind/DefaultCacheProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.fasterxml.jackson.databind;
+
+import com.fasterxml.jackson.databind.util.LookupCache;
+
+/**
+ * Container for {@link LookupCache} instances to use to override default cache implementations used.
+ * Should only be configured via {@link com.fasterxml.jackson.databind.json.JsonMapper.Builder#cacheProvider(DefaultCacheProvider)}.
+ * 
+ * @since 2.16
+ */
+public class DefaultCacheProvider
```

Comment:
I think it'd be better if Provider class was fully immutable, and only Builder mutable. That way Builder has non-final fields, Provider final.